src/data/list/basic.lean

```diff
@@ -1157,6 +1157,12 @@ theorem nth_len_le : ∀ {l : list α} {n}, length l ≤ n → nth l n = none
 | []       n     h := rfl
 | (a :: l) (n+1) h := nth_len_le (le_of_succ_le_succ h)
 
+theorem nth_eq_nth_of_nth_le_eq_nth_le {l₁ l₂ : list α}
```

Comment:
Could you move this below `nth_eq_none_iff`? I think the next two lemmas are more fundamental.